R/check_file_or_url.R

```diff
@@ -17,18 +18,16 @@
 #' "en")
 check_file_or_url <- function(
   file_or_url,
-  locale = "default"
+  locale = "default",
+  encoding = "latin1"
   ) {
   tryCatch(
     {
-      lines <- readr::read_lines(file_or_url,  n_max = 10)
+      lines <- readLines(con = file_or_url, n = 10,
+                         encoding = encoding)
       keywords <- get_keywords_from_lines(lines)
-      if (!("AXIS-VERSION" %in% keywords)) {
-        stop("File is not a px cube: could not find AXIS-VERSION statement")
-      }
-      if (!('CHARSET=\"ANSI\";' %in% lines)) {
-        stop("File has not the expected ANSI encoding: 
-              file an issue and ask for an extensions to other encodings")
+      if (!("MATRIX" %in% keywords)) {
+        stop("File is not a px cube: could not find mandatory MATRIX statement")
```

Comment:
MATRIX is a mandatory px keyword, while the previously used keyword are not mandatory.